### ISSUE TITLE
Console buffer colors not drawn/restored properly (posterized heavily)

### ISSUE THREAD

**Issue by HamRusTal** [FarGroup/FarManager#873]
### Far Manager version

3.0.6369.0 x86

### OS version

10.0.19045.4894

### Other software

[_Chafa_](https://hpjansson.org/chafa/): [`chafa.exe`](https://hpjansson.org/chafa/releases/static/chafa-1.14.4-1-x86_64-windows.zip)

### Steps to reproduce

1. Download the attached image: [*`Kusya.png`*](https://github.com/user-attachments/assets/18d22268-a929-4f3e-8a9e-71285ccbeea6)
2. Put *`chafa.exe`* to the same directory where the image is (or add it to `PATH`).
3. Run FAR (e.g. without plugins), navigate to the directory with *`Kusya.png`* and *`chafa.exe`*; shrink panels vertically (or just hide them by **`CtrlO`**).
4. Run `chafa.exe Kusya.png`, observe the output, run the same command again, observe the output.
5. Expand panels so that they cover part of the image, e.g. just below the green bow (or re-show them by **`CtrlO`**).
6. (optional) Call e.g. the plugins menu (**`F11`**), cancel it (**`Esc`**), observe the background regions affected by drawing the menu (their width depends on the “ClerType-friendly redraw” option in “Interface settings”).
7. Shrink (or hide) panels again, observe the background regions affected by drawing the panels.

### Expected behavior

Colors of the background are preserved as output, regardless of whether any FAR UI elements were ever drawn over them.

### Actual behavior

In Step 4, the first image output is all posterized heavily, the second image is fine:
<details><summary>Reference image</summary>

![image](https://github.com/user-attachments/assets/4b81c64e-e6ea-414a-b013-d265f0642eb0)</details>

After Step 6 (if taken), the regions affected by drawing the menu are posterized:
<details><summary>With ClearType-friendly redraw</summary>

![image](https://github.com/user-attachments/assets/c9bb3362-941e-4071-a580-3d423ed28c80)</details>

After Step 7, the regions affected by drawing the panels are posterized:
<details><summary>Max damage</summary>

![image](https://github.com/user-attachments/assets/78ce27fa-3726-4375-bffa-c024c0a8f866)</details>

**Note 1:** the leftmost column of the affected full rows becomes black, unlike other white background cells.

**Note 2:** there's no such issue in the Command Prompt e.g. when calling and discarding the History menu (**`F7`**):
<details><summary>With the menu still shown</summary>

![image](https://github.com/user-attachments/assets/dd6840d9-7570-4283-9f10-95b7227d5ac4)</details>

**Comment by johnd0e** [comment]
@HamRusTal 
Make sure that VT checkbox is on in Interface option.
Also for the correct processing you must use latest WT or openconsole.exe

**Comment by HamRusTal** [comment]
> Also note that for the correct processing the latest WT or openconsole.exe required.

@johnd0e Thank you, within `OpenConsole.exe` this scenario works as expected (apart from FAR clock being imprinted to the scrollback buffer but that's a distinct issue and not specific to OpenConsole).

@alabuzhev Thank you for the explanation (and for all your hard work on FAR!). I'll close the Issue as Won't Fix due to a Limitation of the platform.

**Comment by j4james** [comment]
> It does not work in the cases where opaque access is not good enough and we absolutely need to see the colors, e.g. to draw shadows or the grabber.

@alabuzhev Just on the subject of shadows, you may be able to achieve something like that with a [DECCARA](https://vt100.net/docs/vt510-rm/DECCARA.html) operation. That can be used to change the attributes of a cell without affecting the text. So to get a shadow similar to what I've seen in the Far Manager screenshots, you could set the background color to black, and apply a dim attribute to darken the foreground. I don't know if that's exactly the effect you're looking for, but it might be better than nothing.

**Comment by alabuzhev** [comment]
@j4james We try to draw 'real' shadows by merging both bg and fg with 50% transparent black:

![image](https://github.com/user-attachments/assets/b3d0462e-cd0d-4379-bf80-584205fad565)

The 16-color fallback looks worse of course, but it's not the end of the world:

![image](https://github.com/user-attachments/assets/c175e7ef-b471-40fa-8899-9e3115d3654e)

The broken text in these areas is probably more pronounced:

![image](https://github.com/user-attachments/assets/5c26e26b-7ed2-490e-9597-9999b223c7ba)

![image](https://github.com/user-attachments/assets/fe6c0c4a-1641-47e0-8da5-d9a9e12666fd)

If we can keep the text intact it would definitely be better than nothing.
I'll try DECCARA, thank you.

**Comment by j4james** [comment]
> We try to draw 'real' shadows by merging both bg and fg with 50% transparent black:

OK, I don't think we can achieve anything quite as good as that with VT sequences as this point in time, because the dim attribute only applies to the foreground color. Technically there is a mode you can set (not yet implemented in Windows Terminal) that would make it apply to both foreground and background, but that also affects the bold/bright and the blink attributes, which is probably not a desired side effect.

> If we can keep the text intact it would definitely be better than nothing.

Yeah, that part should definitely work. It won't be able to create a shadow through the middle of a DBCS character (I think it'll just apply the attributes to both halves), but it at least won't destroy the text.

Btw, I forgot to mention that you'll likely need to change the [DECSACE](https://vt100.net/docs/vt510-rm/DECSACE) setting before using DECCARA, to get it to apply to a rectangular area rather than a "stream" selection.
